diff --git a/Library/Homebrew/hardware.rb b/Library/Homebrew/hardware.rb
index e056439d5..d615e4d9a 100644
--- a/Library/Homebrew/hardware.rb
+++ b/Library/Homebrew/hardware.rb
@@ -14,6 +14,7 @@ module Hardware
         @optimization_flags ||= {
           native:  arch_flag("native"),
           nehalem: "-march=nehalem",
+          sandybridge: "-march=sandybridge",
           core2:   "-march=core2",
           core:    "-march=prescott",
           armv6:   "-march=armv6",
