diff --git a/Library/Homebrew/extend/ENV/shared.rb b/Library/Homebrew/extend/ENV/shared.rb
index e4a15d299..36b84cf50 100644
--- a/Library/Homebrew/extend/ENV/shared.rb
+++ b/Library/Homebrew/extend/ENV/shared.rb
@@ -255,7 +255,7 @@ module SharedEnvExtension
   # @private
   def effective_arch
     if Homebrew.args.build_bottle? && Homebrew.args.bottle_arch
-      Homebrew.args.bottle_arch
+      Homebrew.args.bottle_arch.to_sym
     else
       Hardware.oldest_cpu
     end
