diff --git a/Library/Homebrew/hardware.rb b/Library/Homebrew/hardware.rb
index 789b34fb9..12a3d32a9 100644
--- a/Library/Homebrew/hardware.rb
+++ b/Library/Homebrew/hardware.rb
@@ -14,6 +14,7 @@ module Hardware
         @optimization_flags ||= {
           native:             arch_flag("native"),
           nehalem:            "-march=nehalem",
+          sandybridge:        "-march=sandybridge",
           core2:              "-march=core2",
           core:               "-march=prescott",
           arm_vortex_tempest: "",
